crates/nargo/src/cli/fs/inputs.rs

```diff
@@ -42,6 +43,23 @@ pub(crate) fn read_inputs_from_file<P: AsRef<Path>>(
     Ok((input_map, return_value))
 }
 
+pub(crate) fn read_inputs_from_cli(
+    inputs: Vec<(String, String)>,
+    abi: &Abi,
+) -> Result<(InputMap, Option<InputValue>), CliError> {
+    if abi.is_empty() {
+        return Ok((BTreeMap::new(), None));
+    }
+
+    let abi_map = &abi.to_btree_map();
```

Comment:
What if we moved this to our `Format` enum so that we encapsulate parsing witness values to that object. We could then have a  `cli.rs` file where we have `impl InputValue { try_from_cli_args(..), .. , .. }` like for `TomlTypes`.